UsageEnvironment/Log.cpp

```diff
@@ -69,29 +79,149 @@ void _Log::OutputToStdout(bool toStdout) {
 
 FILE* _Log::AcquireFile() {
   if( strcmp(logFilePath, "") == 0 ) {
-    if (logToStdout) {
+    if( logToStdout ) {
       return stdout;
     } else {
       return stderr;
     }
   } else {
     FILE* logFileHandle = fopen(logFilePath, "a");
     if( logFileHandle == NULL ) {
-      strcpy(logFilePath, "/tmp/live555.log");
-      logFileHandle = fopen(logFilePath, "a");
+      fprintf(stderr, "Couldn't acquire log number %u.%s\n", logFileNum, strerror(errno));
     }
 
     return logFileHandle;
   }
 }
 
+int _Log::SetLogFileNum(unsigned int logNum) {
+#ifdef LOG_FILE_TMP_DIR
+  string newLogFilePath;
+  if( logNum >= kMaxLogs ) {
+    return -1;
+  }
+  printf("Setting log to %u\n", logNum);
+  logFileNum = logNum;
+  newLogFilePath = string(LOG_FILE_TMP_DIR) + string("/live555-") + to_string(logNum) + string(".log");
+  strcpy(logFilePath, newLogFilePath.c_str());
+  return 0;
+#endif
+  return -1;
+}
+
+void _Log::RollLogs() {
+#ifdef LOG_FILE_TMP_DIR
+  if( logFileNum+1 >= kMaxLogs ) {
+    printf("Rolling logs\n");
+    char logFilePrevPath[PATH_MAX];
+    SetLogFileNum(0);
+    if( remove(logFilePath) < 0 ) {
+      perror("Error removing live555 log file 0 tmp log in " LOG_FILE_TMP_DIR);
+    }
+    strcpy(logFilePrevPath, logFilePath);
+    while (NextLogFile() == 0) {
+      if( rename(logFilePath, logFilePrevPath) ) {
+        perror("Error rolling live555 logs in " LOG_FILE_TMP_DIR);
+      }
+      strcpy(logFilePrevPath, logFilePath);
+    }
+    return;
+  }
+  NextLogFile();
+#endif
+}
+
+int _Log::NextLogFile() {
+  return SetLogFileNum(logFileNum + 1);
+}
+
+int _Log::PreviousLogFile() {
+  return SetLogFileNum(logFileNum - 1);
+}
+
+void _Log::BackupLog(const char * backupFileName) {
+#if defined(LOG_FILE_BACKUP_DIR) && defined(LOG_FILE_TMP_DIR)
+  string backupPath = string(LOG_FILE_BACKUP_DIR) + "/" + string(backupFileName);
+  if( access(backupPath.c_str(), F_OK) == 0 && remove(backupPath.c_str()) < 0 ) {
+    perror("Couldn't remove old backup file");
+    return;
+  }
+
+  printf("Backing up log.\n");
+  syslog(LOG_ERR, "RTSP issue triggered. Backing up log.");
```

Comment:
I'm assuming syslog will go to vObserve? In that case, let's also log the filename that you're writing to (since the client may want to juggle multiple log files).